Warnings when compiling in Visual Studio 2008 (and probably other versions)

After the commit 83fc9a51e72976f7b96f8c20ff1465aace709d17 changed the macro GenDescOfTraceRing, the macro generates the following warning in Visual Studio 2008 when compiled with /W2 or higher:

```
  mps\code\trace.c(432) : warning C4146: unary minus operator applied to unsigned type, result still unsigned
  mps\code\trace.c(785) : warning C4146: unary minus operator applied to unsigned type, result still unsigned
  mps\code\trace.c(872) : warning C4146: unary minus operator applied to unsigned type, result still unsigned
  mps\code\trace.c(1844) : warning C4146: unary minus operator applied to unsigned type, result still unsigned
```

This can be fixed by, for example, adding a typecast to the definition in `locus.h` as follows:
`#define GenDescOfTraceRing(node, tr) PARENT(GenDescStruct, trace, &RING_ELT(GenTrace, traceRing, node)[-(int)tr->ti])`
There may be better solutions though.

--- a/code/locus.h
+++ b/code/locus.h
@@ -97,7 +97,7 @@ extern void GenDescEndTrace(GenDesc gen, Trace trace);
 extern void GenDescCondemned(GenDesc gen, Trace trace, Size size);
 extern void GenDescSurvived(GenDesc gen, Trace trace, Size forwarded, Size preservedInPlace);
 extern Res GenDescDescribe(GenDesc gen, mps_lib_FILE *stream, Count depth);
-#define GenDescOfTraceRing(node, tr) PARENT(GenDescStruct, trace, &RING_ELT(GenTrace, traceRing, node)[-tr->ti])
+#define GenDescOfTraceRing(node, tr) PARENT(GenDescStruct, trace, RING_ELT(GenTrace, traceRing, node) - (tr)->ti)
 
 extern Res ChainCreate(Chain *chainReturn, Arena arena, size_t genCount,
                        GenParam params);

